lib/berkshelf/cached_cookbook.rb

```diff
@@ -15,26 +20,71 @@ def from_store_path(path)
         loaded_cookbooks[path.to_s] ||= from_path(path)
       end
 
+      # Creates a new instance of Berkshelf::CachedCookbook from a path on disk containing
+      # a Cookbook.
+      #
+      # The name of the Cookbook is determined by the value of the name attribute set in
+      # the cookbooks' metadata. If the name attribute is not present the name of the loaded
+      # cookbook is determined by directory containing the cookbook.
+      #
+      # @param [#to_s] path
+      #   a path on disk to the location of a Cookbook
+      #
+      # @raise [IOError] if the path does not contain a metadata.rb or metadata.json file
+      #
+      # @return [Ridley::Chef::Cookbook]
+      def from_path(path)
+        path = Pathname.new(path)
+
+        new(nil, path, nil)
+      end
+
+      def checksum(filepath)
+        Chef::Digester.generate_md5_checksum_for_file(filepath)
+      end
+
       private
 
-        # @return [Hash<String, CachedCookbook>]
+      # @return [Hash<String, CachedCookbook>]
       def loaded_cookbooks
         @loaded_cookbooks ||= {}
       end
     end
 
+    attr_accessor :metadata
+    attr_accessor :path
+    attr_accessor :cookbook_version
+
+    # name and metadata are now ignored and should be removed
```

Comment:
https://github.com/berkshelf/berkshelf#cli-usage

No.

TK should be using mixlib-shellout instead of using it as a library, etc.